youtube_dl/extractor/nhl.py

```diff
@@ -349,3 +354,201 @@ def _real_extract(self, url):
+        ).decode('ascii')
+        oauth_request = sanitized_Request(
+            'https://user.svc.nhl.com/oauth/token?grant_type=client_credentials',
+            data='',
+            headers={
+                'Referer': 'https://www.nhl.com/login',
+                'Accept': 'application/json, text/javascript, */*; q=0.01',
+                'Authorization': 'Basic %s' % authorization,
+                'Content-Type': 'application/json',
+            })
+        oauth_response = self.extractor._download_json(
+            oauth_request,
+            None,  # video_id
+            'Requesting OAuth access token',
+            'Unable to get OAuth access token')
+        return oauth_response['access_token']
+
+    def get_game_data(self, video_id, game_id):
+        game_data_url = 'https://statsapi.web.nhl.com/api/v1/schedule?gamePk=%s&expand=schedule.game.content.media.milestones&expand=schedule.game.content.media.epg&expand=schedule.venue' % game_id
+        return self.extractor._download_json(
+            game_data_url,
+            video_id,
+            'Downloading game data',
+            'Unable to download game data')
+
+    def get_stream_data(self, video_id):
+        stream_data_url = 'https://mf.svc.nhl.com/ws/media/mf/v2.4/stream?contentId=%s&playbackScenario=HTTP_CLOUD_WIRED_WEB&sessionKey=%s&auth=response&format=json&platform=WEB_MEDIAPLAYER&_=%s000'
+        session_key = 'abcdefghijklmnop'
+        timestamp = int(time.time())
+        url = stream_data_url % (video_id, session_key, timestamp)
+        return self.extractor._download_json(url, video_id, 'Downloading stream data', 'Unable to download stream data')
+
+
+class NHLTVIE(InfoExtractor):
+    IE_NAME = 'nhl.com:nhltv'
+    _VALID_URL = r'https?://(?:www\.)?nhl.com/tv/(?P<gameId>\d+)(/[^/]+)*(/(?P<id>\d+))?'
+    _NETRC_MACHINE = 'nhltv'
+    _NHLTV_CACHE = 'nhltv'
+    _TESTS = [{
+        # This is a free video that anyone with an NHL TV login can access
+        'url': 'https://www.nhl.com/tv/2016020321/221-1003765/46561403',
+        'md5': '34d9518c495ebdad947b9723b5a7c9a9',
+        'info_dict': {
+            'id': '46561403',
+            'ext': 'ts',
+            'title': '2016-11-26: Anaheim Ducks @ San Jose Sharks (AWAY feed)',
+            'timestamp': 1480217427,
+        },
+        'skip': 'requires a login to NHL TV, and takes a long time to run as the video is 3h+',
+        'params': {
+            'usenetrc': True,
+            'format': '[width=400]',
+            'hls_use_mpegts': True,
+        }
+    }, {
+        'url': 'https://www.nhl.com/tv/2016020362/221-1003808',
+        'only_matching': True,
+    }, {
+        'url': 'https://www.nhl.com/tv/2016020373',
+        'only_matching': True,
+    }]
+
+    def _real_initialize(self):
+        auth = self._downloader.cache.load(self._NHLTV_CACHE, 'auth')
+        self.api = NHLApi(self, auth)
+        if auth is None:
+            self._login()
```

Comment:
I did not test it much but if it does not actually solve the problem then I don't see much point in it either.